internal/services/containers/kubernetes_cluster_addons.go

```diff
@@ -0,0 +1,1195 @@
+										"addon_profile.0.ingress_application_gateway.0.subnet_id",
+									},
+									ValidateFunc: commonValidate.CIDR,
+								},
+								"subnet_id": {
+									Type:     pluginsdk.TypeString,
+									Optional: true,
+									ConflictsWith: []string{
+										"addon_profile.0.ingress_application_gateway.0.gateway_id",
+										"addon_profile.0.ingress_application_gateway.0.subnet_cidr",
+									},
+									ValidateFunc: subnetValidate.SubnetID,
+								},
+								"effective_gateway_id": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+								"ingress_application_gateway_identity": {
+									Type:     pluginsdk.TypeList,
+									Computed: true,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"client_id": {
+												Type:     pluginsdk.TypeString,
+												Computed: true,
+											},
+											"object_id": {
+												Type:     pluginsdk.TypeString,
+												Computed: true,
+											},
+											"user_assigned_identity_id": {
+												Type:     pluginsdk.TypeString,
+												Computed: true,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+
+					"open_service_mesh": {
+						Type:     pluginsdk.TypeList,
+						MaxItems: 1,
+						Optional: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"enabled": {
+									Type:     pluginsdk.TypeBool,
+									Required: true,
+								},
+							},
+						},
+					},
+					"azure_keyvault_secrets_provider": {
+						Type:     pluginsdk.TypeList,
+						MaxItems: 1,
+						Optional: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"enabled": {
+									Type:     pluginsdk.TypeBool,
+									Required: true,
+								},
+								"secret_rotation_enabled": {
+									Type:     pluginsdk.TypeBool,
+									Default:  false,
+									Optional: true,
+								},
+								"secret_rotation_interval": {
+									Type:         pluginsdk.TypeString,
+									Optional:     true,
+									Default:      "2m",
+									ValidateFunc: containerValidate.Duration,
+								},
+								"secret_identity": {
+									Type:     pluginsdk.TypeList,
+									Computed: true,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"client_id": {
+												Type:     pluginsdk.TypeString,
+												Computed: true,
+											},
+											"object_id": {
+												Type:     pluginsdk.TypeString,
+												Computed: true,
+											},
+											"user_assigned_identity_id": {
+												Type:     pluginsdk.TypeString,
+												Computed: true,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+	}
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		MaxItems: 1,
+		Optional: true,
+		Computed: true,
```

Comment:
FWIW in 3.0 I think this can become un-computed as `kube_dashboard` is no longer enabled by default (or available via the method after Kube 1.19)?

```suggestion
```